utils/joiner/tuplejoiner.cpp

```diff
@@ -42,30 +43,49 @@ TupleJoiner::TupleJoiner(
     const rowgroup::RowGroup& largeInput,
     uint32_t smallJoinColumn,
     uint32_t largeJoinColumn,
-    JoinType jt) :
+    JoinType jt,
+    threadpool::ThreadPool *jsThreadPool) :
     smallRG(smallInput), largeRG(largeInput), joinAlg(INSERTING), joinType(jt),
-    threadCount(1), typelessJoin(false), bSignedUnsignedJoin(false), uniqueLimit(100), finished(false)
+    threadCount(1), typelessJoin(false), bSignedUnsignedJoin(false), uniqueLimit(100), finished(false),
+    jobstepThreadPool(jsThreadPool), _convertToDiskJoin(false)
 {
+    uint i;
+
+    getBucketCount();
+    m_bucketLocks.reset(new boost::mutex[bucketCount]);
+
     if (smallRG.getColTypes()[smallJoinColumn] == CalpontSystemCatalog::LONGDOUBLE)
     {
-        STLPoolAllocator<pair<const long double, Row::Pointer> > alloc;
-        _pool = alloc.getPoolAllocator();
-
-        ld.reset(new ldhash_t(10, hasher(), ldhash_t::key_equal(), alloc));
+        ld.reset(new boost::scoped_ptr<ldhash_t>[bucketCount]);
```

Comment:
The same anti-boost story. Could we replace this with std::unique_ptr?

Reply:
Heh, I don't share your zeal for eliminating boost ptrs. :)  That said, I bet we could use sed to do mass substitutions like boost::scoped_ptr -> std::unique_ptr, and it'll mostly work.